src-tauri/src/data_conversion/distance_conversion.rs

```diff
@@ -12,10 +12,16 @@ use crate::mesh::device::MeshNode;
 * Altitude: in meters above sea level, no conversion needed
 */
 pub fn get_distance(node_1: MeshNode, node_2: MeshNode) -> f64 {
+    let default = 0;
     let node_1_data = node_1.data;
     let node_2_data = node_2.data;
-    let node_1_pos = node_1_data.position.unwrap();
-    let node_2_pos = node_2_data.position.unwrap();
+    let has_node_1_pos = node_1_data.position;
+    let has_node_2_pos = node_2_data.position;
+    if has_node_1_pos.is_none() || has_node_2_pos.is_none() {
```

Comment:
This can be rewritten with a couple of (messy but type-safe) `if-let` statements ([link](https://doc.rust-lang.org/rust-by-example/flow_control/if_let.html)).

Reply:
Also see here for a more simplified syntax https://stackoverflow.com/questions/64293804/how-to-write-multiple-condition-in-if-let-statement

Reply:
Same as above comment - swapped it for another match to check the struct fields.